src/EditorFeatures/CSharpTest/ExtractMethod/MiscTests.cs

```diff
@@ -132,14 +132,17 @@ class A
             view.Selection.Select(new SnapshotSpan(
                 view.TextBuffer.CurrentSnapshot, testDocument.SelectedSpans[0].Start, testDocument.SelectedSpans[0].Length), isReversed: false);
 
-            var callBackService = workspace.Services.GetService<INotificationService>() as INotificationServiceCallback;
+            var callBackService = (INotificationServiceCallback)workspace.Services.GetRequiredService<INotificationService>();
             var called = false;
-            callBackService.NotificationCallback = (t, m, s) => called = true;
+            callBackService.NotificationCallback = (_, _, _) => called = true;
 
             var handler = workspace.ExportProvider.GetCommandHandler<ExtractMethodCommandHandler>(PredefinedCommandHandlerNames.ExtractMethod, ContentTypeNames.CSharpContentType);
 
             handler.ExecuteCommand(new ExtractMethodCommandArgs(view, view.TextBuffer), TestCommandExecutionContext.Create());
 
+            var waiter = workspace.ExportProvider.GetExportedValue<IAsynchronousOperationListenerProvider>().GetWaiter(FeatureAttribute.ExtractMethod);
+            await waiter.ExpeditedWaitAsync();
```

Comment:
💡 Consider making an extension method `ExecuteCommandAsync` which invokes `ExecuteCommand` and then waits for the operation to complete. For waiting for the operation to complete, consider waiting for the background work indicator instead of being feature specific. This would maximize generality of the testing approach.

Reply:
I like this idea too, but I think that can happen in a subsequent PR as I'd also like to see less flaky tests